### ISSUE TITLE
Comment Box not visualizing _existing_ comment records supplied from a DB text field

### ISSUE THREAD

**Issue by aMacDawg** [aptkingston/budibase-comment-box#13]
Expected/desired outcome: As a Budibase user, I have a table that stores 'chat logs' that I would like to use the `Budibase Comment Box` for _visualizing_ these logs as individual chats similar to how the conversation would be experienced by an end-user in a conventional chat/messneger application

Current outcome: no 'comments' or messages are displayed despite a table being connected specifically with a `comment` string/text field that stores (stringified) JSON that conforms to this schema:

```
{
  "message": "This is a sample comment.",
  "email": "user@example.com",
  "name": "John Doe",
  "timestamp": 1633036800000
}
```

Current outcome/test case 2: I attempted to submit a _new_ comment into my connected postgres environment, but upon entering some text and clicking the `post` button, I didn't see any request attempt when simultaneously inspecting the dev console

Edit: clarifying the relational (postgres) DB table - currently I'm storing each "comment" as its own record in a table specifically for storing these comments/messages. The `comment` field maps to my understanding of the model as storing a single stringified JSON in the schema described above, the other fields of this `chat_comment` table are:

`| id | created_at (timestamp) | conversation_id (convo id as written from application logic) | comment (string) |`

I have the comment box nested in a `Data Provider` component that maps from the table and I filter on each `conversation_id` value since this `comment box` component is opened in a modal

**Comment by aMacDawg** [comment]
Okay, this is not a "ready to close and move on" comment just yet, but I was able to answer my own question with additional research.

tl;dr - the JSON strings are actually base64 encoded, and delimited by a `|` (not actually stored as an array of objects, which is uhh an interesting decision)

The `Row ID` seems pretty (annoyingly) stuck on the row id itself (a user _would absolutely never_ want to dynamically assign which column to map to, that would be crazy) so I had to do some weird hacking and wrangling with my own data sandbox. 

Hopefully helpful if/when this plugin ever gets worked on again

**Comment by aptkingston** [comment]
Glad you worked out the data structure! To be totally honest I can't remember why I opted to encode each comment individually. I might have been thinking about future enhancements, where doing it this way meant you'd be able to append comments without first having to read all comments, decode, append and encode again, but I genuinely can't remember. There was probably a reason!

Are you having any other issues displaying comments now that you've worked out the correct structure?

**Comment by aMacDawg** [comment]
thanks @aptkingston I appreciate the reply! I'm glad I was able to figure it out -- I guess my remaining issue I'm wrangling right now is related to the `Row ID` field - right now this seems to expect an integer, and ideally I could reference a foreign_key in one table (loaded in via data provider) and use that as the lookup - however, that `Row ID` field seems to be pretty hard coded to the actual row number of the table/record I'm trying to load

**Comment by aptkingston** [comment]
Nice! That quite often tends to be the case with Budibase - there's probably a way to do anything you want, but it might be a bit hacky. We will definitely focus more on UX and making things easier, because while you and I might be able to find these workarounds, the average user might not be.

**Comment by aMacDawg** [comment]
@aptkingston sweet, in my very uneducated opinion, I would think the solution probably revolves around make that "Row ID" field a more flexible lookup as opposed to requiring an `integer` (that specifically maps to a row number)
